modules/apps/29-fee/types/metadata.go

```diff
@@ -0,0 +1,13 @@
+package types
+
+// MetadataFromVersion attempts to parse the given string into a fee version Metadata,
+// an error is returned if it fails to do so.
+func MetadataFromVersion(version string) (Metadata, error) {
```

Comment:
Hmm, I feel the naming could be improved, but I don't have a good suggestion in mind. I think my hesitation lies around the fact that `Metadata` and `Version` aren't super well defined. Maybe something more explicit could help like:
* `ParseMetadataFromVersionString`
* `UnmarshalMetadataFromVersionString`

Another option is:
```go
var metadata Metadata
err := metadata.Unmarshal(version)
```

I guess the current naming is supposed to mimic `AccAddressFromBech32`

Given that I don't have a concrete suggestion, I'm fine to leave as is if you are happy with it

Reply:
Will leave as is for now 👍

Reply:
Happy to leave as is but I will throw out one more suggestion; `ExtractMetadataFromVersionString`